ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -30,5 +30,338 @@
+        REQUIRE(exists == false);
+        REQUIRE(size == 0);
+
+        // Upload a test file
+        auto metadata = client.UploadFile(
+          file_path, test_defaults::bucket_name, test_defaults::obj_prefix + object_key);
+        REQUIRE(metadata.status().ok());
+
+        objects_iterator =
+          client.ListObjects(test_defaults::bucket_name, gcs::Prefix(test_defaults::obj_prefix));
+
+        REQUIRE(std::distance(objects_iterator.begin(), objects_iterator.end()) ==
+          original_number_of_files + 1);
+
+        // Check the bucket contains the file
+        REQUIRE(conn.object_exists(object_key, exists, size) == 0);
+        REQUIRE(exists);
+        REQUIRE(size != 0);
+
+        // Delete the uploaded file
+        auto dl_metadata =
+          client.DeleteObject(test_defaults::bucket_name, test_defaults::obj_prefix + object_key);
+        REQUIRE(dl_metadata.ok());
+
+        // Check that the file has been deleted
+        objects_iterator =
+          client.ListObjects(test_defaults::bucket_name, gcs::Prefix(test_defaults::obj_prefix));
+
+        REQUIRE(std::distance(objects_iterator.begin(), objects_iterator.end()) ==
+          original_number_of_files);
+
+        REQUIRE(conn.object_exists(object_key, exists, size) == 0);
+        REQUIRE(exists == false);
+        REQUIRE(size == 0);
+    }
+
+    SECTION("Read tests", "[gcp-connection]")
+    {
+        // Check number of files with the given prefix currently in bucket
+        auto objects_iterator =
+          client.ListObjects(test_defaults::bucket_name, gcs::Prefix(test_defaults::obj_prefix));
+        int original_number_of_files =
+          std::distance(objects_iterator.begin(), objects_iterator.end());
+
+        // Upload a test file
+        auto metadata = client.UploadFile(
+          file_path, test_defaults::bucket_name, test_defaults::obj_prefix + file_name);
+        REQUIRE(metadata.status().ok());
+
+        objects_iterator =
+          client.ListObjects(test_defaults::bucket_name, gcs::Prefix(test_defaults::obj_prefix));
+
+        REQUIRE(std::distance(objects_iterator.begin(), objects_iterator.end()) ==
+          original_number_of_files + 1);
+
+        // Check the bucket contains the file
+        metadata = client.GetObjectMetadata(
+          test_defaults::bucket_name, test_defaults::obj_prefix + file_name);
+        REQUIRE(metadata.ok());
+
+        int len;
+        int offset;
+
+        // Read GCP objects under the test bucket with no offset.
+        len = 15;
+        offset = 0;
+        void *buf = calloc(1000, sizeof(char));
+
+        REQUIRE(conn.read_object(file_name, offset, len, buf) == 0);
+        REQUIRE(static_cast<char *>(buf) == payload);
+        memset(buf, 0, 1000);
+
+        // Read GCP objects under the test bucket with offset.
+        len = 11;
+        offset = 4;
+
+        REQUIRE(conn.read_object(file_name, offset, len, buf) == 0);
```

Comment:
I don't think we need offset and len as a variable, i prefer us to directly just place the values inside the call.